pkg/models/task_duplicate.go

```diff
@@ -0,0 +1,166 @@
+// Vikunja is a to-do list application to facilitate your life.
+// Copyright 2018-present Vikunja and contributors. All rights reserved.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public Licensee as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public Licensee for more details.
+//
+// You should have received a copy of the GNU Affero General Public Licensee
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+package models
+
+import (
+	"code.vikunja.io/api/pkg/web"
+
+	"xorm.io/xorm"
+)
+
+type TaskDuplicate struct {
+	// The task id of the task to duplicate
+	TaskID    int64 `json:"-" param:"taskid"`
+	ProjectID int64 `json:"-" param:"projectid"`
+
+	// The copied task
+	Task *Task `json:"duplicated_task,omitempty"`
+
+	web.Rights   `json:"-"`
+	web.CRUDable `json:"-"`
+}
+
+// Create duplicates of a task
+// @Summary Duplicate an existing task
+// @Description Copies the task, assignees, lables, subtasks from one task to a new one. The user needs read and write access in the project.
+// @tags task
+// @Accept json
+// @Produce json
+// @Security JWTKeyAuth
+// @Param taskID path int true "The task ID to duplicate"
+// @Param projectID path int true "The project ID of task to duplicate"
+
+// @Success 201 {object} models.TaskDuplicate "The created task."
+// @Failure 400 {object} web.HTTPError "Invalid task duplicate object provided."
+// @Failure 403 {object} web.HTTPError "The user does not have access to the project."
+// @Failure 500 {object} models.Message "Internal error"
+// @Router /projects/{projectID}tasks/{taskID}/duplicate [put]
+func (td *TaskDuplicate) Create(s *xorm.Session, doer web.Auth) (err error) {
+	// Helper to copy a single task (without subtasks)
+	copyTask := func(orig *Task) *Task {
```

Comment:
Why the function?

You could either do it without the function or use `mergo` (third party package already used in other places).